### PR TITLE
feat: support custom container handler class for @Secret

### LINKED ISSUE #81: Allow custom secret container handling
__Is your feature request related to a problem? Please describe.__

I want to hold secrets in a container class that is not handled by the SDK as I need it. If I have secrets somewhere in a JSON object, I would like to replace them in the JSON directly.

__Describe the solution you'd like__

I can add a handler class to the `@Secret` annotation. This handler allows to define how to traverse the container and extract elements to inspect for secrets. From the handler, I can call a `handleElement` method that takes care of the element as it's done in the standard SDK routine.

__Describe alternatives you've considered__

* I can leave out the annotation for my object and use the secret store from the context object to do secret replacement manually. I would however like to reuse the existing replacement capabilities.
* I can transform the JSON to a Java Map and annotate that map for holding secrets. This includes unnecessary transformations though. I might have to transform the map back to JSON afterwards, for example.

__Additional context__

This might be needed in the REST Connector.

### PR DESCRIPTION
**Pull request by tmetzke**
## Description

Allows defining a custom container handler for Secret containers in the `@Secret` annotation.

## Related issues

closes #81

### CODE REVIEW

File: core/src/main/java/io/camunda/connector/impl/ReflectionHelper.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+ */
+package io.camunda.connector.impl;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.Method;
+import java.lang.reflect.Modifier;
+import java.util.Arrays;
+
+/** Utility class for reflection helper methods, like finding getters and setter for fields. */
+public class ReflectionHelper {
```
Comment (igpetrov) [inline core/src/main/java/io/camunda/connector/impl/ReflectionHelper.java:25]: Consider making it final

```suggestion
public final class ReflectionHelper {
```

Comment (igpetrov) [reply core/src/main/java/io/camunda/connector/impl/ReflectionHelper.java:25]: Also, this class is quite complex. Could we add some unit tests for it?

Comment (tmetzke) [reply core/src/main/java/io/camunda/connector/impl/ReflectionHelper.java]: It's actually tested quite extensively in the [OutboundConnectorContextTest](https://github.com/camunda/connector-sdk/pull/83/files#diff-263b4930ecc0ffdb06fbacf3f15dd947b3beddd989030662f97b9bfb755af1c0)

File: core/src/main/java/io/camunda/connector/impl/secret/SecretHandler.java [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+      handleSecretsArray((Object[]) input);
+    } else if (input instanceof Map) {
+      handleSecretsMap((Map<Object, Object>) input);
+    } else if (input instanceof List) {
+      handleSecretsList((List<Object>) input);
+    } else if (input instanceof Iterable) {
+      handleSecretsIterable((Iterable<?>) input);
+    } else {
+      handleSecretsField(input);
+    }
```
Comment (igpetrov) [inline core/src/main/java/io/camunda/connector/impl/secret/SecretHandler.java:71]: Do we really need to handle separately array, list and iterable? Feels like we could simplify this branching.

File: core/src/main/java/io/camunda/connector/impl/ReflectionHelper.java (hunk 1 above)
Comment (igpetrov) [reply core/src/main/java/io/camunda/connector/impl/ReflectionHelper.java]: Thanks for confirming.

**Review by igpetrov — APPROVED** [review]
LGTM

**Review by igpetrov — APPROVED** [review]
Will this also solve https://github.com/camunda/connector-http-json/issues/16?

**Comment by tmetzke** [comment]
> Will this also solve [camunda/connector-http-json#16](https://github.com/camunda/connector-http-json/issues/16)?

Not per se, I think. But it will make it easier to handle. You can for example provide a custom secret handling for JSON content now. If there is enough demand, we could also provide it from the SDK as a convenience.

**Pull request merged**